dbms/src/Server/Server.cpp

```diff
@@ -1158,9 +1158,10 @@ int Server::main(const std::vector<std::string> & /*args*/)
 
     // Initialize the thread pool of storage before the storage engine is initialized.
     LOG_FMT_INFO(log, "dt_enable_read_thread {}", global_context->getSettingsRef().dt_enable_read_thread);
+    // `DMFileReaderPool` should be destructed after `SegmentReaderPoolManager`.
```

Comment:
The comment seems strange. Do you mean `DMFileReaderPool` should be constructed before `SegmentReaderPoolManager`?